backtester/funding/spotpair.go

```diff
@@ -0,0 +1,147 @@
+package funding
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/shopspring/decimal"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+)
+
+var (
+	// ErrNotCollateral is returned when a user requests collateral pair details when it is a funding pair
+	ErrNotCollateral = errors.New("not a collateral pair")
+)
+
+// BaseInitialFunds returns the initial funds
+// from the base in a currency pair
+func (p *SpotPair) BaseInitialFunds() decimal.Decimal {
+	return p.base.initialFunds
+}
+
+// QuoteInitialFunds returns the initial funds
+// from the quote in a currency pair
+func (p *SpotPair) QuoteInitialFunds() decimal.Decimal {
+	return p.quote.initialFunds
+}
+
+// BaseAvailable returns the available funds
+// from the base in a currency pair
+func (p *SpotPair) BaseAvailable() decimal.Decimal {
+	return p.base.available
+}
+
+// QuoteAvailable returns the available funds
+// from the quote in a currency pair
+func (p *SpotPair) QuoteAvailable() decimal.Decimal {
+	return p.quote.available
+}
+
+// Reserve allocates an amount of funds to be used at a later time
+// it prevents multiple events from claiming the same resource
+// changes which currency to affect based on the order side
+func (p *SpotPair) Reserve(amount decimal.Decimal, side order.Side) error {
+	switch side {
+	case order.Buy:
```

Comment:
Maybe add in Bid and ask types for people who make custom strategies? Let me know what you think?